integration/templates/astro-node/src/components/page-with-user.tsx

```diff
@@ -0,0 +1,17 @@
+import React, { type PropsWithChildren, useSyncExternalStore } from 'react';
+import { UserProfile } from '@clerk/astro/client/react';
+import { $userStore } from '@clerk/astro/client/stores';
+
+export function PageWithUser({ children }: PropsWithChildren) {
+  const user = useSyncExternalStore($userStore.listen, $userStore.get, $userStore.get);
```

Comment:
Could we use the [custom store hook](https://github.com/clerk/javascript/blob/d006499c28a493573443d53900a84ba564b61922/packages/astro/src/client/react/utils.tsx#L11) that was previously created for this? Or is the intentional?

Reply:
We are not exporting that in user land tho. I would expect people that use `@clerk/astro` to create their own hook or use `@nanostores/react`. In the future we could have `useUser` exported from `@clerk/astro`